### PR TITLE
payload conversion in progress

### PR DESCRIPTION
**Pull request by roysanchez98**
Working on conversion of payloads parameters.
- Creation of enum for acceleration (byte to float conversion)
- TODO add remaining parameters and conversions (Gyroscope, Magnetosensor)

### CODE REVIEW

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 1]
```diff
@@ -31,50 +31,196 @@
...
-     * accel_Z from accelerometer 
+     * acceleration in the z direction from accelerometer 
+     * Units is meters per second
+     * @param getAccel_Z returns acceleration of z axis
+     * @param setAccel_Z takes acceleration as arg and sets private var 
      */
-    private Short accel_Z;
-    public Short getAccel_Z() { return accel_Z; }
-    public void setAccel_Z(Short accel_Z) { this.accel_Z = accel_Z; }
+    private float accel_Z;
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: Data types must be their object types--use Float

Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: Label the getters and setters independently 

Using document directives should follow this format: `@directive [label] <comment>`

Also, `@param` should be reserved for function parameters

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 2]
```diff
@@ -31,50 +31,196 @@
...
+    	this.mag_X = mag_x * fs_xl.conversionFactor;
+    	this.mag_Y = mag_y * fs_xl.conversionFactor;
+    	this.mag_Z = mag_z * fs_xl.conversionFactor;
     	this.temp_C = temp_c;
     }
     
+    /**
+     * Method to print out all payload attributes in a JSON format
+     */
     @Override
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java:201]: Use of getters is overly verbose

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 3]
```diff
@@ -31,50 +31,196 @@
...
+    	this.gyro_X = gyro_x * fs_xl.conversionFactor;
+    	this.gyro_Y = gyro_y * fs_xl.conversionFactor;
+    	this.gyro_Z = gyro_z * fs_xl.conversionFactor;
+    	this.mag_X = mag_x * fs_xl.conversionFactor;
+    	this.mag_Y = mag_y * fs_xl.conversionFactor;
+    	this.mag_Z = mag_z * fs_xl.conversionFactor;
     	this.temp_C = temp_c;
     }
     
+    /**
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java:198]: Don't start comments with "Method that..." or Function that...", it is implied.

Also, avoid use of "print" unless specifically using any of the Print/Writers/Logger/Builder objects. Instead use "generates"

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 4]
```diff
@@ -31,50 +31,196 @@
...
+    	public final float conversionFactor;
+    	
+    	private FS_G(float convFact) {
+    		this.conversionFactor = convFact / 1000;
+    	}
+    }
+    
+    /**
+     * Determines the scale and precision of the magnetosensor
+     * Conversion to float from conversion factor of enum.
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: Somewhat vague, first comment is sufficient

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 5]
```diff
@@ -31,50 +31,196 @@
...
+    	public final float conversionFactor;
+    	
+    	private FS_XL(float convFact) {	
+    		this.conversionFactor = convFact * 9.80665f / 1000;
+    	}	
+    }
+    
+    /**
+     * Determines the scale and precision of the gyroscope 
+     * Conversion to float from conversion factor of enum.
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: Somewhat vague, first comment is sufficient

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 6]
```diff
@@ -31,50 +31,196 @@
...
     
+    
+    /**
+     * Creates static instance of enum 
+     */
+    public static final FS_XL fs_xl = FS_XL.FS_XL_2;
+    
+    /**
+     *  Determines the scale and precision of the accelerometer
+     *  g is the standard acceleration due to Earth's gravity
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: First comment is sufficient

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 7]
```diff
@@ -31,50 +31,196 @@
...
+     * @param getGyro_X return angular motion of x axis
+     * @param setGryo_X takes ang motion as arg and sets private var  
+     */
+    private float gyro_X;
+    public float getGyro_X() { return gyro_X; }
+    public void setGyro_X(float gyro_X) { this.gyro_X = gyro_X; }
+    
+    /**
+     * angular motion in the y direction from gyroscope
+     * Units is millidegrees per second
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: Grammar: "units *are*", not "is"

**Review by UselessMnemonic — CHANGES_REQUESTED** [review]

File: src/main/java/com/tritoncubed/launchapi/Payload.java (hunk 1 above)
Comment (roysanchez98) [reply src/main/java/com/tritoncubed/launchapi/Payload.java:59]: > Data types must be their object types--use Float

The data types were changed to be float, please re-review.

File: src/main/java/com/tritoncubed/launchapi/DBServlet.java [hunk 8]
```diff
@@ -39,7 +39,7 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 		
 		if(buffer.capacity() != 22) { 
 			response.setStatus(400); 
-			response.getWriter().append("{\"error\" : \"Payload is not 22 bytes. Got " + buffer.capacity() + "\" }" ); 
+			response.getWriter().append("{\"error\" : \"Payload is not 22pay bytes. Got " + buffer.capacity() + "\" }" );
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/DBServlet.java]: Typo

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 9]
```diff
@@ -32,103 +32,114 @@
 
     /**
      * acceleration in the x direction from accelerometer 
-     * Units is meters per second
-     * @param getAccel_X returns acceleration of x axis
-     * @param setAccel_X takes acceleration as arg and sets private var
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: `@directive` is not a document directive

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 10]
```diff
@@ -32,103 +32,114 @@
...
-     * @param getAccel_X returns acceleration of x axis
-     * @param setAccel_X takes acceleration as arg and sets private var 
+     * Units are meters per second
+     * @directive  getAccel_X returns acceleration of x axis
+     * @directive  setAccel_X takes acceleration as arg and sets private var 
      */
     private float accel_X;
-    public float getAccel_X() { return accel_X; }
-    public void setAccel_X(float accel_X) { this.accel_X = accel_X; }
+    Float accel_x = accel_X;
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: Do not add new fields, make the existing fields Float instead

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 11]
```diff
@@ -200,24 +207,24 @@ private FS_M(float convFact) {
      * @param mag_z magnetic field in the z direction
      * @param temp_c temperature from payload
      */
-    public Payload(short accel_x, short accel_y, short accel_z, 
-    short gyro_x, short gyro_y, short gyro_z, short mag_x, 
-    short mag_y, short mag_z, float temp_c) {
+    public Payload(short accel_xx, short accel_yy, short accel_zz,
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java]: Anti-pattern, do not purposefully rename variables if there are scope conflicts

**Review by UselessMnemonic — CHANGES_REQUESTED** [review]
Do not use Acceleration scaling factor for Gyroscope or Magnetosensor readings, use the enums you made

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 12]
```diff
@@ -31,50 +31,179 @@
...
-    private float temp_C;
-    public float getTemp_C() { return temp_C; }
-    public void setTemp_C(float temp_C) { this.temp_C = temp_C; }
+    private Float temp_C;
+    public Float getTemp_C() { return temp_C; }
+    public void setTemp_C(Float temp_C) { this.temp_C = temp_C; }
+    
+    
+    /**
+     * Creates static instance of enum
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java:116]: Configure conversion factors

File: src/main/java/com/tritoncubed/launchapi/Payload.java [hunk 13]
```diff
@@ -31,50 +31,179 @@
...
      * @param accel_x acceleratoin on x axis
      * @param accel_y acceleratoin on y axis 
      * @param accel_z acceleration on z axis
+     * @param gyro_x angular motion in the x direction
+     * @param gyro_y angular motion in the y direction
+     * @param gyro_z angular motion in the z direction
+     * @param mag_x magnetic field in the x direction
+     * @param mag_y magnetic field in the y direction
+     * @param mag_z magnetic field in the z direction
      * @param temp_c temperature from payload
```
Comment (UselessMnemonic) [inline src/main/java/com/tritoncubed/launchapi/Payload.java:180]: From probe

**Review by UselessMnemonic — APPROVED** [review]

**Pull request merged**